File: Java/392.java
class Solution{
    public boolean isSubsequence(String s, String t){
    
        // Solution 1

        // base case
        if (s.length() <= 0){
            return true;
        }

        if (t.length() < s.length()){
            return false;
        }

        int subs = 0;

        for (int i = 0; i <= t.length()-1; i++){
            if (subs <= s.length()-1){
                if (s.charAt(subs) == t.charAt(i)){
                    subs++;
                }
            }
        }

        if (s.length() == subs){
            return true;
        }

        return false;

    }
}

File: Python/392.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:

        # Solution 1
        # using two-pointers

        # base case
        if len(s) < 1:
            return True
        
        if len(s) > len(t):
            return false

        i, j = 0, 0

        while i < len(s) and j < len(t):
            if s[i] == t[j]:
                i += 1
            j += 1

        if i == len(s):
            return True
        else:
            False

        # Solution 2
        # using counts

        # base case
        if len(s) < 1:
            return True;

        if len(s) > len(t):
            return False;

        subs = 0

        for i in range(len(t)):
            if subs <= len(s):
                if s[subs] == t[i]:
                    subs += 1

        if len(s) == subs:
            return True;

        return False;
